ios/Classes/coins/XTZ.swift

```diff
@@ -27,5 +27,132 @@ class XTZ: Coin  {
+                        revealOperation.storageLimit = txData["storageLimit"] as! Int64
+                        revealOperation.kind = .reveal
+                        revealOperation.revealOperationData = revealOperationData
+
+                    operationList.operations = [ transactionOperation, revealOperation ]
+                }else{
+                operationList.operations = [ transactionOperation ]
+                }
+                
+
+                let input = TezosSigningInput.with {
+                    $0.operationList = operationList
+                    $0.privateKey = privateKey.data
+                }
+
+                let output: TezosSigningOutput = AnySigner.sign(input: input, coin: .tezos)
+
+                txHash = output.encoded.hexString
+            case "FA12": 
+                var revealed = txData["revealed"] as! Bool
+                var operationList = TezosOperationList()
+                operationList.branch = txData["branch"] as! String
+                
+                let transactionOperationData = TezosTransactionOperationData.with {
+                    $0.amount = txData["transactionAmount"] as! Int64
+                    $0.destination = txData["destination"] as! String
+                    $0.parameters.fa12Parameters.entrypoint = "transfer";
+                    $0.parameters.fa12Parameters.from = txData["senderAddress"] as! String;
+                    $0.parameters.fa12Parameters.to   = txData["toAddress"] as! String;
+                    $0.parameters.fa12Parameters.value = txData["value"] as! String;
+                }
+
+                let transactionOperation = TezosOperation.with {
+                    $0.source = txData["source"] as! String;
+                    $0.fee = txData["fee"] as! Int64
+                    $0.counter = txData["counter"] as! Int64
+                    $0.gasLimit = txData["gasLimit"] as! Int64
+                    $0.storageLimit = txData["storageLimit"] as! Int64
+                    $0.kind = .transaction
+                    $0.transactionOperationData = transactionOperationData
+                }
+
+                if(revealed == false){
+                    var revealOperationData = TezosRevealOperationData()
+                    revealOperationData.publicKey = publicKey
+
+                    var revealOperation = TezosOperation()
+                        revealOperation.source = txData["source"] as! String;
+                        revealOperation.fee = txData["fee"] as! Int64
+                        revealOperation.counter = txData["counter"] as! Int64
+                        revealOperation.gasLimit = txData["gasLimit"] as! Int64
+                        revealOperation.storageLimit = txData["storageLimit"] as! Int64
+                        revealOperation.kind = .reveal
+                        revealOperation.revealOperationData = revealOperationData
+
+                    operationList.operations = [ transactionOperation, revealOperation ]
+
```

Comment:
fee, counter, gasLimit and storageLimit are different for reveal so have their versions.